morphic.js

```diff
@@ -7818,10 +7824,40 @@ MenuMorph.prototype.select = function (aMenuItem) {
     this.selection = aMenuItem;
 };
 
+MenuMorph.prototype.mouseLeave = function () {
+    if (this.parent instanceof MenuMorph) {
+        // this is a submenu so we vanish when lose focus
+		this.destroy();
+    }
+}
+            
+
 MenuMorph.prototype.destroy = function () {
```

Comment:
something about this method seems overly complicated. What's all the scanning for? Can't you just destroy the menu and be done with? Why does this code need to be extended at all?